library/java/src/io/envoyproxy/envoymobile/engine/EnvoyNativeResourceRegistry.java

```diff
@@ -0,0 +1,72 @@
+package io.envoyproxy.envoymobile.engine;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.ReferenceQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.Set;
+
+import io.envoyproxy.envoymobile.engine.EnvoyNativeResourceReleaser;
+import io.envoyproxy.envoymobile.engine.EnvoyNativeResourceWrapper;
+
+/**
+ * Central class to manage releasing native resources when wrapper objects are flagged as
+ * unreachable by the garbage collector.
+ */
+public enum EnvoyNativeResourceRegistry {
```

Comment:
Is there a reason for this to be an enum?

Reply:
Just a trivial way to make this a singleton for now.

Reply:
Alright